adapter/redisquota/redisquota.go

```diff
@@ -111,34 +114,66 @@ func (rq *redisQuota) alloc(args adapter.QuotaArgs, bestEffort bool) (adapter.Qu
 	amount, exp, err := rq.common.CommonWrapper(args, func(d *adapter.QuotaDefinition, key string, currentTime time.Time, currentTick int64) (int64, time.Time,
 		time.Duration) {
 		result := args.QuotaAmount
-		conn, _ := rq.redisPool.get()
-		// TODO: propagate Connection Pool error here
-		// if err != nil {
-		// 	rq.logger.Infof("Could not get connection to redis")
-		// 	return 0, time.Time{}, 0
-		//}
+		seconds := int32((d.Expiration + time.Second - 1) / time.Second)
+
+		conn, err := rq.redisPool.get()
+		if err != nil {
+			rq.redisError = rq.common.Logger.Errorf("Could not get connection to redis %v", err)
+			return 0, time.Time{}, 0
+		}
 		defer rq.redisPool.put(conn)
 
 		// increase the value of this key by the amount of result
 		conn.pipeAppend("INCRBY", key, result)
-		resp, _ := conn.pipeResponse()
-		// TODO: propagate Connection Pool error here
-		// if err != nil {
-		//	rq.logger.Infof("Could not get response from redis")
-		//	return 0, time.Time{}, 0
-		//}
-		ret := resp.int()
+		resp, err := conn.pipeResponse()
+		if err != nil {
+			rq.redisError = rq.common.Logger.Errorf("Could not get response from redis %v", err)
+			return 0, time.Time{}, 0
+		}
+		ret, err := resp.int()
+		if err != nil {
```

Comment:
If the previous line only returns `err,` these two lines can be combined. Otherwise, if I combine these two lines and use `ret` later, it will complain about `ret` not defined.